### ISSUE TITLE
In the advanced search, add "Rated OK by at least 1 member: Any / Yes / No"

### ISSUE THREAD

**Issue by ckjpn** [Tatoeba/tatoeba2#2261]
## The Enhancement

In the advanced search, add "Rated OK by at least 1 member: Any / Yes / No"

## Why?

So people can limit searches by OK ratings to find potentially good sentences.

And, so people can find sentences that do not yet have OK ratings that they can proofread and add the OK rating to.

**Comment by PaulPeer** [comment]
> Rated OK by at least 1 member: Any / Yes / No"

I would suggest "Rated OK by at least 2 members: Any / Yes / No" because I notice that more and more users start to OK their own sentences... 

> And, so people can find sentences that do not yet have OK ratings that they can proofread and potentially add the OK rating to.

And, although maybe a bit off topic here: I would love a feature that shows my sentences that have been marked as not OK. If users just put the red exclamation mark, but don't give a comment, we might never find out.

**Comment by ckjpn** [comment]
> I would suggest "Rated OK by at least 2 members: Any / Yes / No" because I notice that more and more users start to OK their own sentences...

Good idea.  Possibly even let the person searching decide on the threshold (1, 2, 3, 4, etc.)

> I would love a feature that shows my sentences that have been marked as not OK. 

Perhaps something like this would work, having 2 drop-down option selections.

Rated [OK / Unsure / Not OK] by at least [1 / 2 / 3 / 4] members.

**Comment by ckjpn** [comment]
Maybe more options would be needed, with the defaults being the first ones.

Rated [Any rating or No rating / Not rated / OK / Unsure / Not OK] by at least [0 / 1 / 2 / 3 / 4] members.

It would likely make things more complicated, but the following might also be useful.

Rated [Any rating or No rating / Not rated / OK / Unsure / Not OK]  by at least [0 / 1 / 2 / 3 / 4]  [native speaker / native or non-native speaker / non-native speaker ] members.

If these ideas were to be considered, they would need to be fine-tuned a bit.

**Comment by jiru** [comment]
Thank you for all your suggestions about adding more options, but could we focus more on the use cases instead? It looks like what you are describing is a kind of proofreading activity. We could make a dedicated page for that instead of stuffing more options into the already-cramped advanced search.

@ckjpn If I understand correctly, your real intention is that more contributors put reviews on unreviewed sentences so that more sentences are trustworthy. Is that correct?

@PaulPeer You are pointing out another problem: reviews without comments may go unnoticed. Now how do we solve this problem? If we create a page that list sentences of mine that are marked as not OK, a responsible user like you would certainly check the page on a regular basis, however what about all the members who don’t? Wouldn’t be better to be notified as well?

**Comment by alanfgh** [comment]
Since we allow people to rate sentences as OK, I can see why it would seem
appealing to offer this option. But I actually think it's a really bad
idea, a further step toward the idea of "Distrust every sentence unless CK
marks it as OK." The fact of the matter is that the vast, vast majority of
sentences are fine. This includes sentences by non-native speakers, and I
know what I'm talking about: I've reviewed sentences by all English
speakers with 20 or more sentences who do not self-identify as natives. The
sentences that have issues can and should be corrected. Anything that
pushes us toward a "single-gatekeeper" model (which we will de facto have,
since no one other than CK has the time or inclination to review vast
quantities of sentences) acts against the diversity and quality of our
corpus. And I don't want people to feel obligated to enter an arms race
where they feel the need to review their own sentences as OK just so they
will be seen.

The other thing is that CK fairly often falsely marks sentences, including
his own, as OK. This seems to be due to carelessness rather than problems
with mastery of the language, but it's still pokes a major hole in the
argument for supporting a model that depends on marking sentences as
proofread.

**Comment by agrodet** [comment]
As @jiru mentioned, please detail the use case and we can think of a solution. Extracting the needs: 

> So people can limit searches by OK ratings to find potentially good sentences.

I don't think the logical connection is very accurate here. If the concern is "to find potentially good sentences", limiting searched to OK ratings will de facto eliminate good sentences that are not marked as OK.  
Another way to express an "enhancement" about ratings would be to say that ratings, and in particular NOT OK ratings, are not visible on pages other than the sentence page, and therefore they can't help users to avoid bad sentences. Same concern, different "enhancement". Use case, use case,..

>so people can find sentences that do not yet have OK ratings that they can proofread and potentially add the OK rating to.

That's a different topic. Personally, I don't think that's how ratings should be used but it can be discussed. More generally, we always ask people to detail their needs because we are all biased by our own personal way of doing things. I think the content of your sentence is biased by your personal way of using the feature, and others may not have the same vision. But it doesn't necessarily means it's a bad idea.

> And, although maybe a bit off topic here: I would love a feature that shows my sentences that have been marked as not OK. If users just put the red exclamation mark, but don't give a comment, we might never find out.

I think that you're making a very good point, and I encourage you to open a separate issue to detail your idea, and your ideal solution, so we could elaborate on it.

**Comment by alanfgh** [comment]
There are currently three different pathways for indicating whether a sentence is good/bad: 
(1) Leaving a comment (by convention, a comment thread is only started when a sentence needs fixing, not to say "I checked this sentence is fine")
(2) Leaving a tag, particularly an "@" tag (like "\@check", "\@needs native check", or "\@change"), though we also have "OK" tags, and CK uses a "List 907" tag to indicate sentences that meet his personal criteria. I also add a tag like "unnatural" or "not a sentence" when, for one reason or another, asking for modification, performing modification myself, or deleting the sentence are not viable options.
(3) Using the Reviews feature.

I much prefer a combination of (1) and (2). In particular, when I see a sentence that has an issue, and the user is still active, I generally leave a "\@change" tag AND a comment that gives the current content and one or more suggested rewordings, like this:

> The writting is on the wall.
> ->
> The writing is on the wall.

The "\@change" tag ensures that it won't get lost later. Not only such sentences with this tag be found through a custom advanced search, but the "Improve sentences" link from the user interface links to a search that is already set up for "\@change", "\@check", and "\@needs native check". (It also points to a search for sentences with the "OK" tag, which I don't think is particularly useful.)

It feels like the desire to have searches be able to discriminate between sentences that have been assigned a "review" (via the Reviews feature) is an attempt to achieve something that is already available via the use of tags, and this will only lead to confusion: some people will use the Reviews feature (despite its limited nature), while others will use tags, with the result that a corpus maintainer can no longer depend on one method to look for sentences that require attention.

**Comment by agrodet** [comment]
@alanfgh, you're making good points, as usual. Two little remarks : 

I think that the "List 907" tag was used as a replacement of the list, but since lists are running now, the tag could be deleted. I even think that CK asked for massive removal. The fact that we don't have a U.I. for removing / editing tags is already reported in another issue. 

We shouldn't have to choose, or enforce a choice on users, between 1, 2, 3 or a mix of any of them. What we lack, and we are fully aware of it, is an overall proofreading / correction system, that would encompass most of our remarks and needs. As an example, it is wrong, from a design point of view, to use tags for corpus proofreading. For many reasons: tags should be used for categorization, not everybody can tag, it is a long process to tag and untag, etc. However, we have to acknowledge the fact that most of our proofreading goes through the use of tags, understand why (because that's the most efficient way of doing it currently), and develop a (or several) better alternative(s). 

The same logic applies to the original issues posted here. Note that Reviews are a fragile feature. If the reviewer is now inactive, outdated reviews will stay forever. Even if reviewers are active, if they didn't let a comment, it might be difficult to see the problem. We also have to acknowledge flaws in the review feature, and try to build a better system overall. 

Finally, may it be for tags, reviews, or other features, a new solution cannot allow confusion such as described at the end of @alanfgh's comment. The fact that two features may or may not overlap isn't necessarily a problem if they are purposely designed that way.

**Comment by jiru** [comment]
@alanfgh Good arguments, I agree with you.

I think this discussion shows we have an intrinsic problem with the ratings feature. Do you remember that this feature were [initially named "users corpus"](https://tatoeba.org/eng/wall/show_message/23719)? It was then [shortly renamed to "collection"](https://tatoeba.org/eng/wall/show_message/23875), and recently to "ratings". I find that interesting because despite the name change, the usage is still assimilated to a "collection" as in "selecting a subset of the corpus depending on your own judgment". It feels like two different ways (one "innocent", and one a bit more "questionable") to look at the same thing.

I didn’t like this feature when it was introduced, and now even more when I see its potential to antagonize members. We should be happily working together towards a common goal.

> Rated [Any rating or No rating / Not rated / OK / Unsure / Not OK] by at least [0 / 1 / 2 / 3 / 4] [native speaker / native or non-native speaker / non-native speaker ] members.

The reviews feature has been left half-finished and I don’t think it should be ever completed as it was initially designed (a kind of scoring system).

> The fact of the matter is that the vast, vast majority of sentences are fine. This includes sentences by non-native speakers, and I know what I'm talking about: I've reviewed sentences by all English speakers with 20 or more sentences who do not self-identify as natives.

I agree, but let’s not forget the situation could be other languages. As a matter of fact, the community of English speakers is active and large, so sentences are getting a lot of attention. To the contrary, languages with less or no active contributors and/or native speakers are more likely to contain errors.

**Comment by ckjpn** [comment]
> I think that the "List 907" tag was used as a replacement of the list, but since lists are running now, the tag could be deleted. I even think that CK asked for massive removal.

https://github.com/Tatoeba/tatoeba2/issues/330#issuecomment-539769850

If someone who has rights to the database can safely take care of this, it would be OK with me.

**Comment by ckjpn** [comment]
Here is a related discussion on the Wall.

https://tatoeba.org/en/wall/show_message/38920#!#message_38920
